src/node_api.h

```diff
@@ -135,6 +135,16 @@ napi_create_external_buffer(napi_env env,
                             void* finalize_hint,
                             napi_value* result);
 #endif  // NODE_API_NO_EXTERNAL_BUFFERS_ALLOWED
+
+#ifndef NODE_API_EXPERIMENTAL_HAS_CREATE_BUFFER_FROM_ARRAYBUFFER
```

Comment:
We must not use the #ifndef like this here. Instead, it must be the #ifdef for the NAPI_EXPERIMENTAL and then we must #define the NODE_API_EXPERIMENTAL_HAS_CREATE_BUFFER_FROM_ARRAYBUFFER.